edx_exams/apps/core/signals/signals.py

```diff
@@ -0,0 +1,21 @@
+from openedx_events.learning.data import ExamAttemptData, UserData, UserPersonalData
+from openedx_events.learning.signals import EXAM_ATTEMPT_SUBMITTED
+
+
+def emit_exam_attempt_submitted_event(user, course_key, usage_key):
+    # .. event_implemented_name: EXAM_ATTEMPT_SUBMITTED
+    EXAM_ATTEMPT_SUBMITTED.send_event(
+        exam_attempt=ExamAttemptData(
+            student_user=UserData(
+                id=user.id,
+                is_active=user.is_active,
+                pii=UserPersonalData(
+                    username=user.username,
+                    email=user.email,
+                    name=user.full_name
+                )
+            ),
+            course_key=course_key,
+            usage_key=usage_key,
```

Comment:
Is there also supposed to be a `requesting_user` passed along? Looking at https://github.com/openedx/openedx-events/blob/fec19895eed1c37e4da6305bfa9849665a832372/openedx_events/learning/data.py#L316

Reply:
Or I guess it defaults to None, and only a learner can submit their own exam

Reply:
I don't think so. The plan was to reserve this attribute for cases where a user other than the learner has made the change, like an instructor resetting an exam in the Instructor Dashboard or overriding an exam attempt status on manual review, for example. I don't believe that submission can be triggered by anyone but the learner.

Reply:
I went back on this and decided to add in the `requesting_user` whenever there is one. I would prefer to be explicit about the `requesting_user` than having an implicit understanding that it should be empty whenever the learner is the requester.